src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -2313,6 +2313,14 @@ def generate(
                 .repeat(batch_size, 1)
                 .to(image_embeds.device)
             )
+            if getattr(self.config, "image_token_index", None) is not None:
+                image_token_ids = (
+                    torch.LongTensor([[self.config.image_token_index]])
```

Comment:
```suggestion
                    torch.tensor([[self.config.image_token_index]], device=image_embeds.device, dtype=torch.long)
```
otherwise this is the legacy constructor !
```python
In [3]: torch.LongTensor(1, device="mps")
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[3], line 1
----> 1 torch.LongTensor(1, device="mps")

RuntimeError: legacy constructor expects device type: cpu but device type: mps was passed
```
and let's split in 2 lines 😉